src/mpi/attr/attrutil.c

```diff
@@ -242,10 +239,12 @@ int MPIR_Attr_delete_list(int handle, MPIR_Attribute ** attr)
         }
         /* --END ERROR HANDLING-- */
         /* For this attribute, find the delete function for the
-         * corresponding keyval */
-        /* Still to do: capture any error returns but continue to
-         * process attributes */
-        mpi_errno = MPIR_Call_attr_delete(handle, p);
+         * corresponding keyval. If the delete function fails,
+         * we record the last failure */
+         rc = MPIR_Call_attr_delete(handle, p);
+         if (rc != 0) {
```

Comment:
This this trivial change we can report the first failure.

```suggestion
         if (mpi_errno == MPI_SUCCESS && rc != 0) {
```

Reply:
The suggestion sounds okay